frontend/src/RecordingPageComp/RecordingPage.jsx

```diff
@@ -1,21 +1,43 @@
 import { useContext } from "react";
 import { UserContext } from "../UserContext.js";
-import { useState } from "react";
+import { useState, useRef, useEffect } from "react";
 import { Link } from "react-router-dom";
+import MIDISounds from "midi-sounds-react";
 import NavBar from "../NavBarComp/NavBar.jsx";
 import RecordingComp from "./RecordingComp.jsx";
 import SimilarSounds from "./SimilarSounds.jsx";
 import PlaybackContainer from "./PlaybackContainerComp.jsx";
+import InstrumentSelection from "./InstrumentSelectionComp.jsx";
 import "./RecordingPage.css";
 
 function RecordingPage() {
   const { user } = useContext(UserContext);
   const [note, setNote] = useState(0);
   const [playbackDuration, setPlaybackDuration] = useState(0);
+  const [selectedInstrument, setSelectedInstrument] = useState("");
+  const midiSounds = useRef(null);
+
+  const [isMidiReady, setIsMidiReady] = useState(false);
+  useEffect(() => {
+    const checkMidiInitialization = () => {
+      if (midiSounds.current && midiSounds.current.player) {
+        setIsMidiReady(true);
+      }
+    };
+    setTimeout(checkMidiInitialization, 1000); 
```

Comment:
Clean up the timer after unmount? Just incase any side effect
timer = setTimeout(checkMidiInitialization, 1000);
return () => clearTimeout(timer)